### PR TITLE
CORE: fix ucc parser build

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
## What
Align ucc_config_parser_fill_opts with changes in UCS 
https://github.com/openucx/ucx/pull/8557

## Why ?
Fixes UCC build with latest UCX

### CODE REVIEW

**Review by manjugv — APPROVED** [review]

File: src/utils/ucc_parser.c [hunk 1]
```diff
@@ -316,20 +316,24 @@ static ucc_status_t ucc_apply_file_cfg(void *opts, ucc_config_field_t *fields,
...
+ucc_status_t ucc_config_parser_fill_opts(void *opts, ucs_config_global_list_entry_t *entry,
+                                         const char *env_prefix, int ignore_errors)
 {
     ucs_status_t ucs_status;
     ucc_status_t status;
 
-    ucs_status = ucs_config_parser_fill_opts(opts, fields, env_prefix,
-                                             table_prefix, ignore_errors);
-    status     = ucs_status_to_ucc_status(ucs_status);
+#if UCS_HAVE_CONFIG_GLOBAL_LIST_ENTRY_FLAGS
```
Comment (shimmybalsam) [inline src/utils/ucc_parser.c:325]: Perhaps change to UCC macro instead of UCS

File: src/utils/ucc_parser.h [hunk 2]
```diff
@@ -65,10 +87,8 @@ typedef struct ucc_config_names_list {
     };
 } ucc_config_names_list_t;
 
-ucc_status_t ucc_config_parser_fill_opts(void *opts, ucc_config_field_t *fields,
-                                         const char *env_prefix,
-                                         const char *table_prefix,
-                                         int         ignore_errors);
+ucc_status_t ucc_config_parser_fill_opts(void *opts, ucs_config_global_list_entry_t *entry,
```
Comment (shimmybalsam) [inline src/utils/ucc_parser.h:90]: alignment line too long

File: test/gtest/utils/test_cfg_file.cc [hunk 3]
```diff
@@ -61,8 +66,9 @@ class test_cfg_file : public ucc::test {
     void init_cfg() {
         ucc_status_t status;
         std::swap(ucc_global_config.file_cfg, file_cfg);
-        status = ucc_config_parser_fill_opts(&cfg, ucc_gtest_config_table,
-                                             "GTEST_UCC_", "CFG_", 1);
+        status = ucc_config_parser_fill_opts(&cfg,
+                                             UCC_CONFIG_GET_TABLE(ucc_gtest_config_table),
```
Comment (shimmybalsam) [inline test/gtest/utils/test_cfg_file.cc:70]: alignment line too long

**Review by shimmybalsam — APPROVED** [review]

File: src/utils/ucc_parser.c (hunk 1 above)
Comment (Sergei-Lebedev) [reply src/utils/ucc_parser.c:326]: prefix is UCS because it depends on libucs similarly to UCS_HAVE_MPOOL_PARAMS

**Comment by artemry-nv** [comment]
bot:retest

**Comment by artemry-nv** [comment]
bot:retest

**Comment by artemry-nv** [comment]
bot:retest

**Review by bureddy — APPROVED** [review]

**Pull request merged**